### PR TITLE
Cookiecutter the library and update to match NeoPixel API.

### PR DESCRIPTION
**Pull request by tannewt**

### CODE REVIEW

**Comment by tannewt** [comment]
Yeah, I want our libraries to have the adafruit_ prefix. neopixel is unique since its more prevalent already.

File: adafruit_dotstar.py [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+    :param int n: The number of dotstars in the chain
+    :param float brightness: Brightness of the pixels between 0.0 and 1.0
+    :param bool auto_write: True if the neopixels should immediately change when set. If False, `show` must be called explicitly.
+
+
+    Example for Gemma M0:
+
+    .. code-block:: python
+
+        import dotstar
```
Comment (dhalbert) [inline adafruit_dotstar.py]: `import adafruit_dotstar`

File: adafruit_dotstar.py [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+        import time
+        from board import *
+
+        RED = 0x100000
+
+        with dotstar.DotStar(APA102_SCK, APA102_MOSI, 1) as pixels:
```
Comment (dhalbert) [inline adafruit_dotstar.py]: `with adafruit_dotstar.DotStar(...)`

**Comment by dhalbert** [comment]
Got it re naming. So a couple of examples have to change in dotstar.py. Tnx.

**Comment by tannewt** [comment]
Fixed up the example and deinit code. Tested by @kattni and I on the CircuitPlayground Express with DotStar strips.

**Pull request merged**